x/staking/keeper/msg_server_test.go

```diff
@@ -1350,6 +1353,20 @@ func (s *KeeperTestSuite) TestConsKeyRotn() {
 			errMsg:    "exceeding maximum consensus pubkey rotations within unbonding period",
 			validator: validators[5].GetOperator(),
 		},
+		{
+			name: "try using the old pubkey of another validator that rotated",
+			malleate: func() sdk.Context {
+				val, err := stakingKeeper.ValidatorAddressCodec().StringToBytes(validators[0].GetOperator())
+				s.Require().NoError(err)
+
+				bankKeeper.EXPECT().SendCoinsFromAccountToModule(gomock.Any(), sdk.AccAddress(val), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+				return ctx
+			},
+			isErr:     true,
+			errMsg:    "validator already exist for this pubkey; must use new validator pubkey",
+			newPubKey: validator0PubKey,
+			validator: validators[2].GetOperator(),
+		},
```

Comment:
👍 good test case. 

We are only missing a scenario where the key was rotated in the current block already to cover the `rotationHistory`.

Reply:
That was done here: https://github.com/cosmos/cosmos-sdk/pull/20649 👌